tutorial/markdown/java/spring-boot.md

```diff
@@ -73,152 +122,261 @@ Once the site is up and running you can launch your browser and go to the Swagge
-```
+## Route Specifications
 
-> *from controllers/ProfileController.java*
+### GET Route
 
-## GET a Profile by Key
+`@GetMapping("/{id}")`
+This route is used to retrieve an airline by its unique identifier (ID).
 
-Navigate to the getProfile method in the ProfileController file in the controllers folder.  We only need the profile ID `pid` or our `Key` from the user to retrieve a particular profile document using a basic key-value operation which is passed in the method signature as a string.  Since we created the document with a unique key we can use that key to find the document in the scope and collection it is stored in.
+- It expects the id of the airline as a path parameter.
+- It calls the `getAirlineById` method of the `AirlineService` to retrieve the airline with the specified ID.
+- If the airline is found, it returns a `ResponseEntity` with HTTP status `200 OK` and the airline data in the response body.
+- If the airline is not found, it returns a `ResponseEntity` with HTTP status `404 Not Found`.
+- If any other error occurs, it returns a `ResponseEntity` with HTTP status `500 Internal Server Error`.
 
-```java
-Profile profile = profileCol.get(pid).contentAs(Profile.class);
-return ResponseEntity.status(HttpStatus.OK).body(profile);
-```
+### POST Route
 
-> *from getProfile method of controllers/ProfileController.java*
+`@PostMapping("/{id}")`
+This route is used to create a new airline.
 
-## PUT Profile
+- It expects the id of the airline as a path parameter, but this ID is typically generated by the server.
+- It receives the airline data in the request body, which should be a valid JSON representation of an airline.
+- It calls the `createAirline` method of the `AirlineService` to create the new airline.
+- If the airline is created successfully, it returns a `ResponseEntity` with HTTP status `201 Created` and the created airline data in the response body.
+- If a conflict occurs (e.g., an airline with the same ID already exists), it returns a `ResponseEntity` with HTTP status `409 Conflict`.
+- If any other error occurs, it returns a `ResponseEntity` with HTTP status `500 Internal Server Error`.
 
-Update a Profile by Profile ID
+### PUT Route
 
-Now let's navigate to the update method of the ProfileController class.  The entire document gets replaced except for the document key and the `pid` field.  We create a call to the `collection` using the `upsert` method and then return the document saved and the result just as we did in the previous endpoint.
+`@PutMapping("/{id}")`
+This route is used to update an existing airline by its ID.
 
-```java
-profileCol.upsert(id, profile);
-return ResponseEntity.status(HttpStatus.CREATED).body(profile);
-```
+- It expects the id of the airline as a path parameter.
+- It receives the updated airline data in the request body.
+- It calls the `updateAirline` method of the `AirlineService` to update the airline with the specified ID.
```

Comment:
Can you also add the Couchbase/Spring API used by the endpoint with a link to docs for all these end points?

Reply:
linked to relevant parts of the sdk docs